tests/test_project.py

```diff
@@ -144,12 +144,13 @@ def recursive_update(d, u):
             config = project.config.copy()
             config = recursive_update(config, config_overrides)
             project = project_class(config=config)
-        for a in range(3):
+        for a in range(2):
             if heterogeneous:
                 # Add jobs with only the `a` key without `b`.
                 project.open_job(dict(a=a)).init()
                 project.open_job(dict(a=dict(a=a))).init()
-            for b in range(3):
+            # Tests assume that there are even and odd values of b
+            for b in range(2):
```

Comment:
I discussed this change with @vyasr. It should be fine for us to reduce the number of jobs from 9 to 4, by reducing the number of values for `a` and `b`. I scanned the tests for any place where this might reduce coverage but I don't see any issues.